compiler/backend/ccgexprs.nim

```diff
@@ -1534,12 +1536,16 @@ proc genOrd(p: BProc, e: CgNode, d: var TLoc) =
 proc genSomeCast(p: BProc, e: CgNode, d: var TLoc) =
   const
     ValueTypes = {tyTuple, tyObject, tyArray, tyOpenArray, tyVarargs, tyUncheckedArray}
+
+  let src =
+    if e.kind == cnkCall: e[1]
+    else:                 e.operand
```

Comment:
`genSomeCast` is also used to generate the code for `mChr`, which is why this dispatching is needed here.